pkg/controller/windowsmachine/metrics/metrics.go

```diff
@@ -0,0 +1,219 @@
+}
+
+// Configure configures patches the endpoint object to reflect the current list Windows nodes.
+func (pc *prometheusNodeConfig) Configure() error {
+	// Check if metrics are enabled in current cluster
+	if !metricsEnabled {
+		log.Info("install the prometheus-operator to enable prometheus configuration")
+		return nil
+	}
+	// get list of Windows nodes that are in Ready phase
+	nodes, err := pc.k8sclientset.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{LabelSelector: nodeconfig.WindowsOSLabel,
+		FieldSelector: "spec.unschedulable=false"})
+	if err != nil {
+		return errors.Wrap(err, "could not get Windows nodes")
+	}
+
+	// get Metrics Endpoints object
+	endpoints, err := pc.k8sclientset.CoreV1().Endpoints(pc.namespace).Get(context.TODO(),
+		windowsMetricsEndpoints, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "could not get metrics endpoints %v", windowsMetricsEndpoints)
+	}
+
+	if !pc.isEndpointsValid(nodes, endpoints) {
+		// check if we can get list of endpoint addresses
+		windowsIPList := pc.getNodeEndpointAddresses(nodes)
+		log.Info("get ip list", "list", windowsIPList)
+		// sync metrics endpoints object with the current list of addresses
+		if err := pc.syncMetricsEndpoint(windowsIPList); err != nil {
+			return errors.Wrap(err, "error updating endpoints object with list of endpoint addresses")
+		}
+	}
+	log.Info("prometheus configured", "endpoints", windowsMetricsEndpoints, "port", Port, "name", PortName)
+	return nil
+}
+
+// getNodeEndpointAddresses returns a list of endpoint addresses according to the given list of Windows nodes
+func (pc *prometheusNodeConfig) getNodeEndpointAddresses(nodes *v1.NodeList) []v1.EndpointAddress {
+	// an empty list to store node IP addresses
+	var nodeIPAddress []v1.EndpointAddress
+	// loops through nodes
+	for _, node := range nodes.Items {
+		for _, address := range node.Status.Addresses {
+			if address.Type == "InternalIP" && address.Address != "" {
+				// add IP address address to the endpoint address list
+				nodeIPAddress = append(nodeIPAddress, v1.EndpointAddress{
+					IP:       address.Address,
+					Hostname: "",
+					NodeName: nil,
+					TargetRef: &v1.ObjectReference{
+						Kind: "Node",
+						Name: node.Name,
+					},
+				})
+				break
+			}
+		}
+	}
+	// return list and error
+	return nodeIPAddress
+}
+
+// isEndpointsValid returns true if Endpoints object has entries for all the Windows nodes in the cluster
+// else returns false.
+func (pc *prometheusNodeConfig) isEndpointsValid(nodes *v1.NodeList, endpoints *v1.Endpoints) bool {
+	// check if number of entries in endpoints object match number of Ready Windows nodes
+	if len(nodes.Items) != len(endpoints.Subsets[0].Addresses) {
+		return false
+	}
+
+	for _, node := range nodes.Items {
+		nodeFound := false
+		for _, address := range endpoints.Subsets[0].Addresses {
+			if address.TargetRef.Name == node.Name {
+				nodeFound = true
+				break
```

Comment:
Can't you just `return true` here? With that you won't need `nodeFound`.

Reply:
I want to return `true` only after all nodes are found and return if `single` node is not found

Reply:
Please call that out explicitly in the doc then.